src/Symfony/Component/Security/Csrf/Tests/TokenGenerator/UriSafeTokenGeneratorTest.php

```diff
@@ -19,42 +19,46 @@
  */
 class UriSafeTokenGeneratorTest extends TestCase
 {
-    private const ENTROPY = 1000;
-
     /**
-     * A non alpha-numeric byte string.
-     *
-     * @var string
+     * @dataProvider tokenDataProvider
      */
-    private static $bytes;
+    public function testGenerateToken(string $format)
+    {
+        $generator = new UriSafeTokenGenerator(1000);
+        $token = $generator->generateToken();
 
-    /**
-     * @var UriSafeTokenGenerator
-     */
-    private $generator;
+        $this->assertTrue(ctype_print($token), 'is printable');
+        $this->assertStringNotMatchesFormat($format, $token, 'is URI safe');
+    }
 
-    public static function setUpBeforeClass(): void
+    public static function tokenDataProvider(): \Iterator
     {
-        self::$bytes = base64_decode('aMf+Tct/RLn2WQ==');
+        yield ['%S+%S'];
+        yield ['%S/%S'];
+        yield ['%S=%S'];
     }
 
-    protected function setUp(): void
+    /**
+     * @dataProvider validDataProvider
+     */
+    public function testValidLength(int $entropy, int $length)
     {
-        $this->generator = new UriSafeTokenGenerator(self::ENTROPY);
+        $generator = new UriSafeTokenGenerator($entropy);
+        $token = $generator->generateToken();
+        $this->assertSame($length, \strlen($token));
     }
 
-    protected function tearDown(): void
+    public static function validDataProvider(): \Iterator
     {
-        $this->generator = null;
+        yield [24, 4];
+        yield 'Float length' => [20, 3];
     }
 
-    public function testGenerateToken()
+    public function testInvalidLength()
     {
-        $token = $this->generator->generateToken();
+        $this->expectException(\InvalidArgumentException::class);
+        $this->expectExceptionMessage('Entropy should be greater than 0.');
 
-        $this->assertTrue(ctype_print($token), 'is printable');
-        $this->assertStringNotMatchesFormat('%S+%S', $token, 'is URI safe');
-        $this->assertStringNotMatchesFormat('%S/%S', $token, 'is URI safe');
-        $this->assertStringNotMatchesFormat('%S=%S', $token, 'is URI safe');
```

Comment:
why removing these test cases? can you please minimize the diff on this file to ease the review, or explain the changes otherwise?

Reply:
Actually, I didn't delete the test; I just modified it. The code that I removed was the code below. `$bytes` is not used, and I thought the remaining code was unnecessary because it's just a simple test. The diff should be fine now.

```php
private const ENTROPY = 1000;

/**
 * A non alpha-numeric byte string.
 *
 * @var string
 */
private static $bytes;

/**
 * @var UriSafeTokenGenerator
 */
private $generator;

public static function setUpBeforeClass(): void
{
    self::$bytes = base64_decode('aMf+Tct/RLn2WQ==');
}

protected function setUp(): void
{
    $this->generator = new UriSafeTokenGenerator(self::ENTROPY);
}

protected function tearDown(): void
{
    $this->generator = null;
}
```